MiniDumper/MiniDumper.cs

```diff
@@ -178,6 +180,28 @@ public void PrintDebugString(OUTPUT_DEBUG_STRING_INFO outputDbgStrInfo)
             }
         }
 
+        public void MemoryCommitThreshold(int commitThreshold)
+        {
+            System.Timers.Timer aTimer = new System.Timers.Timer();
+            aTimer.Interval = 1000;
+            aTimer.Elapsed += (sender, e) => OnTimedEvent(sender, e, commitThreshold); ;
+            aTimer.AutoReset = true;
+            aTimer.Enabled = true;
+        }
+
+        private void OnTimedEvent(Object source, ElapsedEventArgs e, int commitThreshold)
+        {
+           var process = Process.GetProcessById(pid);
+            double privateMemory = (process.PrivateMemorySize64 / 1024) / 1024;
```

Comment:
I would prefer to convert the command line parameter to bytes (`MemoryCommitThresholdInMB * 1024 * 1024` and assign to a `UInt64` variable) and later compare it with the process memory usage - this way we are not loosing the last bytes of the gathered value. 

Also please use the `GetProcessMemoryInfo` method to query the process memory:

```
[StructLayout(LayoutKind.Sequential)]
private struct PROCESS_MEMORY_COUNTERS_EX
{
	public uint cb;
	public uint PageFaultCount;
	public UIntPtr PeakWorkingSetSize;
	public UIntPtr WorkingSetSize;
	public UIntPtr QuotaPeakPagedPoolUsage;
	public UIntPtr QuotaPagedPoolUsage;
	public UIntPtr QuotaPeakNonPagedPoolUsage;
	public UIntPtr QuotaNonPagedPoolUsage;
	public UIntPtr PagefileUsage;
	public UIntPtr PeakPagefileUsage;
	public UIntPtr PrivateUsage;
}

[DllImport("psapi.dll", SetLastError = true)]
static extern bool GetProcessMemoryInfo(IntPtr hProcess, out PROCESS_MEMORY_COUNTERS_EX counters, uint size);

EXAMPLE:

	PROCESS_MEMORY_COUNTERS_EX counters;
	counters.cb = (uint)Marshal.SizeOf<PROCESS_MEMORY_COUNTERS_EX>();
	var result = GetProcessMemoryInfo(processHandle, out counters, counters.cb);
	if (!result) {
		// TODO throw exception with last error
	} 
``` 

The `PrivateUsage` field in the structure is the one that interests us - use `ToUInt64` method to convert it to a number.